### ISSUE TITLE
Accessibility issue smartmenus

### ISSUE THREAD

**Issue by create-martin** [moodle-an-hochschulen/moodle-theme_boost_union#694]
**Describe the bug**
Using smart menus, the aria info aria-current="true" is not set correctly.
So they fail in standard accessibility tests.

Since we have a lot of clients, that need to be as accessible as possible (there's actually some laws coming into effect in 2025 in Europe) this is quite troublesome.

**To Reproduce**
Just create smartmenu --> e.g. just put a static page/dashbaord behind the link. aria-current is always on "dashbaord"

**Expected behavior**
having aria-current "moved" to correct place

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by wiebkemueller-hsh** [comment]
Hello @create-martin 

thank you for highlighting this and also the legal implications there are in EU law. We have two questions:

- Is the behaviour you are expecting only "behind the scenes" (the aria info aria-current="true") or is this also tied to a visual recognition to be made by the user? We've been talking about that in a ticket that should (if yes) then be linked to this issue.

- You seem to already have had a very good insight in the details of the problem. Are you in any way able to provide a PR that could be integrated? We're still running short on DEV resources and help would be very much appreciated!

Regards on behalf of the DEV team!
Wiebke

**Comment by create-martin** [comment]
Hi Wiebke,

I was on vacation, hence the delay in answering.
As for the first question: to pass a11y tests, the "behind-the-scences" is probably enough. From a user experience point of view, I would very much like it, if the current menu option can also be visibly different from the other ones (using core, that would just be the underline hightlight; everything else can be done via theming anyway I suspect? is there a CSS class for it? or one could probably use the correctly-set-aria-label for it???). 

As for developing ressources. I will check, please give me a couple of days to find out

**Comment by abias** [comment]
Hi all,

as I have just commented in #715 and #620:

The (visual) highlighting of the active smart menu node will be fixed by #648 soon.
The change will be backported to Moodle 4.4 and 4.3, but not to 4.1.
With this patch, the active node will also get the `aria-current="true"` attribute.

@create-martin , I would appreciate if you could verify (after #648 is integrated) that the behaviour matches your expectations. If yes, you can close this issue. If no, please let us know what is still missing.

Cheers,
Alex

**Comment by abias** [comment]
@create-martin 
#648 has been integrated and I have verified that the active node gets the `aria-current="true"` attribute.

Against this background I close this issue for now.

Cheers,
Alex
